istioctl/pkg/proxystatus/proxystatus.go

```diff
@@ -129,6 +132,17 @@ Retrieves last sent and last acknowledged xDS sync from Istiod to each Envoy in
 				if err != nil {
 					return err
 				}
+				pod, err := kubeClient.Kube().CoreV1().Pods(ns).Get(context.TODO(), podName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				annotations := klabels.Set(pod.ObjectMeta.Annotations)
+				opts.Revision = describe.GetRevisionFromPodAnnotation(annotations)
```

Comment:
This will lead to the original flag not working as expected. Also, what if the annotaion is not the same as the control plane?

Reply:
Thanks for attention, I think:
1. This annotation is injected into envoy by istiod, and there will be no situation where this annotation is inconsistent with the revision of istiod
2. "This will lead to the original flag not working as expected.": In the case of `istioctl ps <podName>`, the `--revision` parameter is meaningless, because if the `revision` of the pod is inconsistent with the `revision` of isitod, normal output cannot be obtained.

Reply:
@hanxiaop I made some improvements to the `--revision` handling. Could you review it again?